### PR TITLE
auth0 and netlify auth packages

### PR DESCRIPTION
**Pull request by Tobbe**
Move auth0 and netlify into their own packages nested in `packages/auth-providers`

### CODE REVIEW

**Comment by jtoar** [comment]
@Tobbe think I fixed build in https://github.com/redwoodjs/redwood/pull/6922/commits/b961af1f13f1381e6448b9b1a9cb8c0c7ed0b0d9; had to use a glob in the `workspaces` field in the root package.json. `yarn workspaces list` now shows the nested packages:

```
dom@evaM1 ~/p/r/redwood (tobbe-auth-by-provider) [1]> yarn workspaces list 
➤ YN0000: .
➤ YN0000: packages/api
➤ YN0000: packages/api-server
➤ YN0000: packages/auth
➤ YN0000: packages/auth-providers-api
➤ YN0000: packages/auth-providers-setup
➤ YN0000: packages/auth-providers-web
➤ YN0000: packages/auth-providers/auth0/api
➤ YN0000: packages/auth-providers/auth0/setup
➤ YN0000: packages/auth-providers/auth0/web
➤ YN0000: packages/auth-providers/netlify/api
➤ YN0000: packages/auth-providers/netlify/setup
➤ YN0000: packages/auth-providers/netlify/web
➤ YN0000: packages/cli
➤ YN0000: packages/cli-helpers
➤ YN0000: packages/codemods
➤ YN0000: packages/core
➤ YN0000: packages/create-redwood-app
➤ YN0000: packages/eslint-config
➤ YN0000: packages/forms
➤ YN0000: packages/graphql-server
➤ YN0000: packages/internal
➤ YN0000: packages/prerender
➤ YN0000: packages/record
➤ YN0000: packages/router
➤ YN0000: packages/structure
➤ YN0000: packages/telemetry
➤ YN0000: packages/testing
➤ YN0000: packages/web
➤ YN0000: Done in 0s 1ms
```

The Netlify packages don't build yet though, I'm getting:

```
dom@evaM1 ~/p/r/redwood (tobbe-auth-by-provider)> yarn workspace @redwoodjs/auth-netlify-api run build
babel:
  src does not exist
```

There's no src dirs there—was that one simple enough to forego?

File: packages/auth-providers/auth0/api/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  "devDependencies": {
+    "@auth0/auth0-spa-js": "1.22.5",
+    "@babel/cli": "7.19.3",
+    "@babel/core": "7.20.2",
+    "@redwoodjs/auth": "3.2.0",
+    "@redwoodjs/cli-helpers": "3.2.0",
+    "@types/jsonwebtoken": "8.5.9",
+    "@types/react": "17.0.50",
+    "@types/uuid": "8.3.4",
```
Comment (jtoar) [inline packages/auth-providers/auth0/api/package.json:39]: Don't need the types either (haven't looked at the rest of these yet)

```suggestion
```

**Comment by Tobbe** [comment]
That was most likely a copy/paste or drag/drop error on my part. I'll add the src folder back in

File: packages/auth-providers/auth0/api/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    "build:watch": "nodemon --watch src --ext \"js,ts,tsx,template\" --ignore dist --exec \"yarn build\"",
+    "prepublishOnly": "NODE_ENV=production yarn build",
+    "test": "jest src",
+    "test:watch": "yarn test --watch"
+  },
+  "dependencies": {
+    "@babel/runtime-corejs3": "7.20.1",
+    "core-js": "3.26.0",
+    "jsonwebtoken": "8.5.1",
+    "uuid": "9.0.0"
```
Comment (jtoar) [inline packages/auth-providers/auth0/api/package.json:29]: Doesn't seem like we use `uuid` anywhere, but looks like we're missing `jwks-rsa` (I'm just looking at dist to figure out cause it's one file)

```suggestion
    "@babel/runtime-corejs3": "7.20.1",
    "core-js": "3.26.0",
    "jsonwebtoken": "8.5.1",
    "jwks-rsa": "2.0.5"
```

File: packages/auth-providers/auth0/setup/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "prepublishOnly": "NODE_ENV=production yarn build",
+    "test": "jest src",
+    "test:watch": "yarn test --watch"
+  },
+  "dependencies": {
+    "@babel/runtime-corejs3": "7.20.1",
+    "@redwoodjs/auth": "3.2.0",
+    "@redwoodjs/cli-helpers": "3.2.0",
+    "core-js": "3.26.0",
+    "secure-random-password": "0.2.3"
```
Comment (jtoar) [inline packages/auth-providers/auth0/setup/package.json:30]:
```suggestion
    "@babel/runtime-corejs3": "7.20.1",
    "@redwoodjs/cli-helpers": "3.2.0",
    "core-js": "3.26.0"
```

File: packages/auth-providers/auth0/setup/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/** @type {import('@jest/types').Config.InitialOptions} */
+module.exports = {
+  testEnvironment: 'jest-environment-jsdom',
```
Comment (jtoar) [inline packages/auth-providers/auth0/setup/jest.config.js:3]: Probably shouldn't be jsdom cause CLI

```suggestion
```

File: packages/auth-providers/auth0/web/package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "build:types": "tsc --build --verbose",
+    "build:watch": "nodemon --watch src --ext \"js,ts,tsx,template\" --ignore dist --exec \"yarn build\"",
+    "prepublishOnly": "NODE_ENV=production yarn build",
+    "test": "jest src",
+    "test:watch": "yarn test --watch"
+  },
+  "dependencies": {
+    "@babel/runtime-corejs3": "7.20.1",
+    "core-js": "3.26.0",
+    "uuid": "9.0.0"
```
Comment (jtoar) [inline packages/auth-providers/auth0/web/package.json:28]:
```suggestion
    "@babel/runtime-corejs3": "7.20.1",
    "@redwoodjs/auth": "3.2.0",
    "core-js": "3.26.0"
```

File: packages/auth-providers/auth0/web/package.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  "devDependencies": {
+    "@auth0/auth0-spa-js": "1.22.5",
+    "@babel/cli": "7.19.3",
+    "@babel/core": "7.20.2",
+    "@redwoodjs/auth": "3.2.0",
```
Comment (jtoar) [inline packages/auth-providers/auth0/web/package.json:34]:
```suggestion
```

**Review by jtoar — COMMENTED** [review]
Just superficially audited basic things like package.json's, jest.config.js's; will need to do it again more thoroughly for devDeps, possibly peerDeps, etc

File: packages/auth-providers/auth0/api/package.json (hunk 1 above)
Comment (Tobbe) [reply packages/auth-providers/auth0/api/package.json:39]: Done

File: packages/auth-providers/auth0/api/package.json (hunk 2 above)
Comment (Tobbe) [reply packages/auth-providers/auth0/api/package.json:29]: Done

File: packages/auth-providers/netlify/setup/jest.config.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+/** @type {import('@jest/types').Config.InitialOptions} */
+module.exports = {
+  testEnvironment: 'jest-environment-jsdom',
```
Comment (jtoar) [inline packages/auth-providers/netlify/setup/jest.config.js:3]: One last fix:

```suggestion
```

**Review by jtoar — APPROVED** [review]
@Tobbe the last commit I pushed moved a lot of deps/devDeps to peerDeps since these are plugins. Let me know what you think, but this all looks good and I'll look into publishing once merged

File: packages/auth-providers/netlify/web/package.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  "name": "@redwoodjs/auth-netlify-web",
+  "version": "3.2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/redwoodjs/redwood.git",
+    "directory": "packages/auth-providers/netlify/web"
+  },
+  "license": "MIT",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
```
Comment (jtoar) [inline packages/auth-providers/netlify/web/package.json:11]: There's no `dist/index.js` file; this either needs to point to netlify file in dist or we need to add an index like in auth0

File: packages/auth-providers/netlify/setup/package.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "@redwoodjs/auth-netlify-setup",
+  "version": "3.2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/redwoodjs/redwood.git",
+    "directory": "packages/auth-providers/netlify/setup"
+  },
+  "license": "MIT",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
```
Comment (jtoar) [inline packages/auth-providers/netlify/setup/package.json:11]: There's no `dist/index.js` file; this either needs to point to setup file in dist or we need to add an index like in auth0

File: packages/auth-providers/netlify/api/package.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  "name": "@redwoodjs/auth-netlify-api",
+  "version": "3.2.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/redwoodjs/redwood.git",
+    "directory": "packages/auth-providers/netlify/api"
+  },
+  "license": "MIT",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.ts",
```
Comment (jtoar) [inline packages/auth-providers/netlify/api/package.json:11]: There's no `dist/index.js` file; this either needs to point to decoder file in dist or we need to add an index like in auth0

**Review by jtoar — CHANGES_REQUESTED** [review]
Missed one thing about the netlify packages (missing index.js files, the entry points)

**Pull request merged**